semgrep-core/src/analyzing/CFG_build.ml

```diff
@@ -230,51 +233,66 @@ let rec cfg_stmt : state -> F.nodei option -> stmt -> cfg_stmt_result =
        *                 |-----------|-> newfakefinally -> finally
        *
        *)
-      let newi = state.g#add_node { F.n = NOther (Noop "try") } in
+      let newi = state.g#add_node { F.n = Reg (NOther (Noop "try")) } in
       state.g |> add_arc_from_opt (previ, newi);
-      let catchesi = state.g#add_node { F.n = NOther (Noop "catch") } in
+      let catchesi = state.g#add_node { F.n = Reg (NOther (Noop "catch")) } in
       let state' = { state with try_catches_opt = Some catchesi } in
-      let finaltry = cfg_stmt_list state' (Some newi) try_st in
+      let finaltry = cfg_stmt_list lang state' (Some newi) try_st in
       state.g |> add_arc_from_opt (finaltry, catchesi);
-      let newfakefinally = state.g#add_node { F.n = NOther (Noop "finally") } in
+      let newfakefinally =
+        state.g#add_node { F.n = Reg (NOther (Noop "finally")) }
+      in
       state.g |> add_arc (catchesi, newfakefinally);
       catches
       |> List.iter (fun (_, catch_st) ->
-             let finalcatch = cfg_stmt_list state (Some catchesi) catch_st in
+             let finalcatch =
+               cfg_stmt_list lang state (Some catchesi) catch_st
+             in
              state.g |> add_arc_from_opt (finalcatch, newfakefinally));
-      let finalfinally = cfg_stmt_list state (Some newfakefinally) finally_st in
+      let finalfinally =
+        cfg_stmt_list lang state (Some newfakefinally) finally_st
+      in
       (* If we're inside another Try then we assume that we could propagate up
        * some unhandled exception. Otherwise we assume that we handled everything.
        * THINK: Alternatively, we could add an arc to the exit node. *)
       state.g |> add_arc_opt_to_opt (finalfinally, state.try_catches_opt);
       CfgFirstLast (newi, finalfinally)
   | Throw (tok, e) ->
-      let newi = state.g#add_node { F.n = F.NThrow (tok, e) } in
+      let newi = state.g#add_node { F.n = Reg (F.NThrow (tok, e)) } in
       state.g |> add_arc_from_opt (previ, newi);
       (match state.try_catches_opt with
       | None -> state.g |> add_arc (newi, state.exiti)
       | Some catchesi -> state.g |> add_arc (newi, catchesi));
       CfgFirstLast (newi, None)
+  (* Any DefStmts which are FuncDefs are reified as proper Func nodes
+     within the CFG, which contain their own smaller CFGs.
+  *)
+  | MiscStmt (DefStmt (_, G.FuncDef fdef)) ->
```

Comment:
If we are going to build the CFG here, then I think it's better to change the IL so that we add a `DefStmt` to `IL.stmt_kind`. Then AST_to_IL would do the translation.